CMake/SetTargetSourceFolder.cmake

```diff
@@ -0,0 +1,19 @@
+function(set_target_source_folder target)
```

Comment:
That works for me, mostly. The term `set` is still confusing me, because normally I expect a CMake command with `set` to take arguments, whereas this automatically figures that out.